packages/webapp/src/containers/CropVarieties/useCropVarietyCatalogue.js

```diff
@@ -0,0 +1,262 @@
+import {
+  getCurrentManagementPlans,
+  getExpiredManagementPlans,
+  getPlannedManagementPlans,
+} from '../managementPlanSlice';
+import { useSelector } from 'react-redux';
+import { cropCatalogueFilterDateSelector, cropVarietyFilterSelector } from '../filterSlice';
+import { useMemo } from 'react';
+import useStringFilteredCrops from '../CropCatalogue/useStringFilteredCrops';
+import {
+  ACTIVE,
+  COMPLETE,
+  LOCATION,
+  PLANNED,
+  STATUS,
+  SUPPLIERS,
+  NEEDS_PLAN,
+} from '../Filter/constants';
+import { useTranslation } from 'react-i18next';
+import useFilterNoPlan from './useFilterNoPlan';
+import useSortByCropTranslation from '../CropCatalogue/useSortByCropTranslation';
+import { managementPlansWithCurrentLocationSelector } from '../Task/TaskCrops/managementPlansWithLocationSelector';
+import { cropVarietiesSelector } from '../../containers/cropVarietySlice';
+
+export default function useCropVarietyCatalogue(filterString, crop_id) {
+  const managementPlansWithCurrentLocation = useSelector(
+    managementPlansWithCurrentLocationSelector,
+  );
+
+  let cropVarieties = useSelector(cropVarietiesSelector);
+
+  const managementPlansWithCurrentLocationByCropId = managementPlansWithCurrentLocation.filter(
+    (c) => c.crop_id === crop_id,
+  );
+
+  cropVarieties = cropVarieties.filter((c) => c.crop_id === crop_id);
+
+  const cropCatalogFilterDate = useSelector(cropCatalogueFilterDateSelector);
+  let cropCatalogueFilter = useSelector(cropVarietyFilterSelector(crop_id));
+  const managementPlansFilteredByFilterString = useStringFilteredCrops(
+    managementPlansWithCurrentLocationByCropId,
+    filterString,
+  );
+
+  if (!cropCatalogueFilter) {
+    cropCatalogueFilter = {
+      [LOCATION]: {},
+      [SUPPLIERS]: {},
+      [STATUS]: {},
+    };
+  }
+
+  const _without_management_plan = useSortByCropTranslation(useFilterNoPlan(filterString, crop_id));
+
+  const _without_management_plan_ByCropId = _without_management_plan.filter(
+    (c) => c.crop_id === crop_id,
+  );
```

Comment:
I'm not sure of the exact details here, but it seems to me like this could be simplified. On line 53, you are calling `useFilterNoPlan` with the `crop_id`, and then you are filtering it a second time to get all the results with a matching `crop_id`. Can we simplify this to loop over the data fewer times?